### PR TITLE
docker: Upgrade container packages before running tests

### PR DESCRIPTION
**Pull request by mudler**
We need something like this - or either rebuild the docker image more frequently (assuming that it's upgrading packages during docker build phase). 

This is for a later discussion @foursixnine, just opening the PR just to give a rough idea of what we do already in os-autoinst (but there we rebuild the docker image each time, not here, thus tests will fail anyway).

### CODE REVIEW

**Comment by mudler** [comment]
But i would rather drop the cpanm mechanisms to make it fail when trying to upgrade packages, it's rather annoying than useful

File: script/docker-tests [hunk 1]
```diff
@@ -14,13 +15,17 @@ function prepare_fullstack_test {
 
     git clone https://github.com/os-autoinst/os-autoinst.git ../os-autoinst
     cd ../os-autoinst
-    cpanm -n --mirror http://no.where/ --installdeps .
+    [ "$INSTALL_FROM_CPAN" -eq 1 ] && \
+	      (cpanm --local-lib=~/perl5 local::lib && cpanm -n --installdeps . ) || \
+	      cpanm -n --mirror http://no.where/ --installdeps .
     if ! [ $? -eq 0 ]; then
```
Comment (foursixnine) [inline script/docker-tests]: in this case, if INSTALL_FROM_CPAN is not set, the exit of the cd that is previous is what's evaluated right?... I think we can remove then the check

File: docker/openqa/entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    [ "$INSTALL_FROM_CPAN" -eq 1 ] && eval $(perl -I ~/perl5/lib/perl5/ -Mlocal::lib) || true
+}
+
+export -f create_db run_as_normal_user
+
+run_as_normal_user;
+
+echo ">> Running tests"
+
+dbus-run-session -- sh -c "$*"
```
Comment (foursixnine) [inline docker/openqa/entrypoint.sh]: :+1:

File: script/docker-tests (hunk 1 above)
Comment (mudler) [reply script/docker-tests]: right, will address this in another PR

**Pull request merged**